autosklearn/util/data.py

```diff
@@ -14,6 +25,96 @@
 from autosklearn.evaluation.splitter import CustomStratifiedShuffleSplit
 
 
+# TODO: TypedDict with python 3.8
+#
+#   When upgrading to python 3.8 as minimum version, this should be a TypedDict
+#   so that mypy can identify the fields types
+DatasetCompressionSpec = Dict[str, Union[float, List[str]]]
+
+# Default specification for arg `dataset_compression`
+default_dataset_compression_arg:  DatasetCompressionSpec = {
+    "memory_allocation": 0.1,
+    "methods": ["precision", "subsample"]
+}
+
+
+def validate_dataset_compression_arg(
+    dataset_compression: Mapping[str, Any],
+    memory_limit: int
+) -> DatasetCompressionSpec:
+    """Validates and return a correct dataset_compression argument
+
+    The returned value can be safely used with `reduce_dataset_size_if_too_large`.
+
+    Parameters
+    ----------
+    dataset_compression: Mapping[str, Any]
+        The arg to validate
+
+    Returns
+    -------
+    DatasetCompressionSpec
+        The validated and correct dataset compression spec
+    """
+    if isinstance(dataset_compression, Mapping):
+        # Fill with defaults if they don't exist
+        dataset_compression = {
+            **default_dataset_compression_arg,
+            **dataset_compression
+        }
+
+        # Must contain known keys
+        if set(dataset_compression.keys()) != set(default_dataset_compression_arg.keys()):
+            raise ValueError(
+                f"Unknown key in dataset_compression, {list(dataset_compression.keys())}."
+                f"\nPossible keys are {list(default_dataset_compression_arg.keys())}"
+            )
+
+        memory_allocation = dataset_compression["memory_allocation"]
+
+        # "memory_allocation" must be float or int
+        if not (isinstance(memory_allocation, float) or isinstance(memory_allocation, int)):
+            raise ValueError("key 'memory_allocation' must be an `int` or `float`")
```

Comment:
Could you please print the illegal type, too?

Reply:
Yus.